detectable/src/main/java/com/synopsys/integration/detectable/detectables/yarn/YarnTransformer.java

```diff
@@ -176,20 +176,23 @@ private void collectYarnDependencies(
                     dependencyVersion = dependency.getVersion();
                 }
                 LazyId stringDependencyId = generateComponentDependencyId(dependency.getName(), dependencyVersion);
-                if (yarnDependencyTypeFilter.shouldInclude(YarnDependencyType.NON_PRODUCTION) || !dependency.isOptional()) {
-                    graphBuilder.setDependencyInfo(stringDependencyId, dependency.getName(), dependencyVersion, generateComponentExternalId(dependency.getName(), dependencyVersion));
-                    //graphBuilder.addChildWithParent(stringDependencyId, id);
-                    LazyDependencyInfo childInfo = graphBuilder.checkAndHandleMissingExternalId(lazyBuilderHandler, stringDependencyId);
-                    Dependency child = new Dependency(childInfo.getName(), childInfo.getVersion(), childInfo.getExternalId(), null);
-                    mutableDependencyGraph.addChildWithParent(child, parent);
-
-                } else {
-                    logger.trace("Excluding optional dependency: {}", stringDependencyId);
-                }
+                moveToMethod(dependency, dependencyVersion, parent,lazyBuilderHandler, graphBuilder, mutableDependencyGraph, stringDependencyId);
             }
         }
     }
 
+    private void moveToMethod(YarnLockDependency dependency, String dependencyVersion, Dependency parent, LazyBuilderMissingExternalIdHandler lazyBuilderHandler, ExternalIdDependencyGraphBuilder graphBuilder, BasicDependencyGraph mutableDependencyGraph,LazyId stringDependencyId) throws MissingExternalIdException {
```

Comment:
@gopannabd Moved last **if** block to a method, please suggest an appropriate name. Also, please review for any issues that may not be obvious to someone without context into this method ... such as whether or not passing the relevant objects around via parameters could cause any issues or not.. etc). In general I lack context into this area of the code, let me know what you think overall. 

Sonar issue: https://sonarcloud.io/project/issues?open=AY36qfWcZptyxDnuTZGc&id=com.synopsys.integration%3Asynopsys-detect

Reply:
includeNonProductionOrOptionalIfNeeded(...)